tests/integration/services/import_services/ocpi/sw_stuttgart/sw_stuttgart_service_test.py

```diff
@@ -0,0 +1,106 @@
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+from decimal import Decimal
+from unittest.mock import ANY, Mock
+
+from tests.integration.services.import_services.ocpi.sw_stuttgart.sw_stuttgart_data import sw_stuttgart_response_json
+from webapp.common import Logger
+from webapp.dependencies import dependencies
+from webapp.models import Connector, Evse, Location
+from webapp.services.import_services.ocpi.sw_stuttgart import SWStuttgartImportService
+
+
+def test_sw_stuttgart_import(
+    app,
+    db,
+    requests_mock,
+):
+    """
+    Test for the SWStuttgartImportService.
+
+    Checks the total number of imported datasets
+    (should be 234 locations, 523 evses and 526 connectors more than before),
+    and a sample of the imported data's content.
+    """
+
+    sw_stuttgart_service: SWStuttgartImportService = dependencies.get_import_services().sw_stuttgart_import_service
+
+    # mock logger
+    sw_stuttgart_service.remote_helper.logger = Mock(Logger)
+
+    locations_in_db_before = db.session.query(Location).count()
+    evses_in_db_before = db.session.query(Evse).count()
+    connectors_in_db_before = db.session.query(Connector).count()
+
+    # define mocked response
+    requests_mock.get(
+        'http://mocked-sw-stuttgart:5000/SW-Stuttgart',
+        status_code=200,
+        json=sw_stuttgart_response_json,
+    )
+
+    # run the import (this should make a request to the mocked sw stuttgart, parse the response, and save the data).
+    sw_stuttgart_service.download_and_save()
+
+    # check numbers of imported objects
+    assert db.session.query(Location).count() - locations_in_db_before == 234
+    assert db.session.query(Evse).count() - evses_in_db_before == 523
+    assert db.session.query(Connector).count() - connectors_in_db_before == 526
+
+    # check a sample of the imported data
+    sample_location = db.session.query(Location).filter(Location.uid == '2185120').first()
+    assert sample_location is not None
+    assert sample_location.to_dict() == {
+        'id': ANY,
+        'created': ANY,
+        'modified': ANY,
+        'uid': '2185120',
+        'source': 'sw_stuttgart',
+        'operator_id': ANY,
+        'suboperator_id': None,
+        'owner_id': None,
+        'dynamic_location_id': None,
+        'dynamic_location_probability': None,
+        'name': 'B+B Tiefgarage / Kronprinzstr. 6 / 4. UG',
+        'address': 'Kronprinzstr. 6 / 4. UG',
+        'postal_code': '70173',
+        'city': 'Stuttgart',
+        'state': None,
+        'country': 'DE',
+        'lat': Decimal('48.7774100'),
+        'lon': Decimal('9.1762120'),
+        'directions': '[{"language": "DE", '
```

Comment:
Hmm. I think it would be better to select the first / only language text there instead of dumping the JSON. Frontends showing this would show the JSON, which is not that nice to look at.

Reply:
OK, gonna do that... I think we can safely assume that there will never be other languages in this field, because SW Stuttgart doesn't stick to the OCPI standard anyway here.

Reply:
Looking into it further... the conversion of this field happens in a generic OCPI mapper, so that would not only affect SW Stuttgart's data, but potentially all OCPI-based data sources. 
So this will lead to problems if we ever want to add an OCPI-based data source that actually has data in multiple languages, where we might want to display more than one of them...